### DISCUSSION TITLE
Disable Copilot on a per-project/per-workspace basis?

### DISCUSSION THREAD

**Discussion by amalyam** [community/community#79257] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I can't seem to find a simple way to disable Copilot on a per-project or per-workspace basis. Would like to have the ability to do this easily in future versions!

**Comment by ayim** [comment]
Take a look at the Content Exclusion feature and let us know what you think.
The team is working to re-introduce this feature across clients in the near future.
Visual Studio users today have the ability to disable Copilot on a per solution or per language basis as well.

**Reply by ayim** [reply]
https://docs.github.com/en/copilot/managing-copilot-business/configuring-content-exclusions-for-github-copilot

**Reply by amalyam** [reply]
Thanks for your reply. The solution in that doc is only for business, not for the individual plan, right? Can you say more about how to disable copilot on a per solution basis in Visual Studio? I'm having trouble finding information about how to do this.

**Reply by ayim** [reply]
Specific to Visual Studio 2022 (not VS Code), there exists an icon in the bottom left of the editor that opens up the menu.
![image](https://github.com/community/community/assets/10951869/e382a3db-780f-48b9-b491-671672951aaf)

Click this, then unclick the "Enable for xxxx.sln".

Note that as we are transitioning towards a new entry point & content exclusion, the items in this menu may change.
Telemetry shows that usage of the disable by solution feature is quite low and usually because the predictions aren't particularly relevant at the moment.

In the new year and moving forward, using .copilotignore or "content exclusion" will be the recommended way to do what you're trying to do.

**Reply by stoerr** [reply]
@ayim The content exclusions that are available in copilot business are a step in the right direction, but way too little for various reasons.
- It applies only if Github Copilot is officially supported by the company. That's a pretty high bar to clear from my experience.
- It doesn't help at all if the repository is hosted somewhere else than Github
Im my experience it's still rare for companies to officially adopt Github Copilot use - many of them are still running "pilot phases" where they explore that for a few people. This missing feature makes it really hard for developers to use it at least in open source projects or private projects when they are also working with proprietary code from the same IntelliJ instance.

This really needs a way to exclude projects and individual modules of a project. Please please please PLEASE!

**Reply by ayim** [reply]
@stoerr thank you for your passion towards content exclusion

RE: non Git hosted repos, you should be able to content exclude them using the instructions here: 
https://docs.github.com/en/copilot/managing-github-copilot-in-your-organization/configuring-content-exclusions-for-github-copilot#configuring-content-exclusions-for-your-organization

RE: company adoption of GitHub Copilot, users have the option of using Copilot for Business by attaching their account to a GitHub organization (free) or GitHub enteprise (paid).

 I'm goign to cc @jenverse on feedback for content exclusion here

**Comment by Mustafa-Hassan2001** [comment]
As of my last knowledge update in January 2022, GitHub Copilot doesn't have a built-in per-project or per-workspace disable feature. Check for updates in future versions for this capability.
